edgelet/hsm-sys/azure-iot-hsm-c/tests/edge_openssl_int/edge_openssl_int.c

```diff
@@ -47,57 +49,101 @@ static char* TEST_IOTEDGE_HOMEDIR_GUID = NULL;
+#define TEST_SERVER_CERT_ECC_FILE_3_NAME "server_ecc_cert_3.cert.pem"
+#define TEST_CLIENT_CERT_ECC_FILE_1_NAME "client_ecc_cert_1.cert.pem"
+static char *TEST_CA_CERT_ECC_FILE_1     = NULL;
+static char *TEST_CA_CERT_ECC_FILE_2     = NULL;
+static char *TEST_SERVER_CERT_ECC_FILE_1 = NULL;
+static char *TEST_SERVER_CERT_ECC_FILE_3 = NULL;
+static char *TEST_CLIENT_CERT_ECC_FILE_1 = NULL;
+
+#define TEST_CA_PK_ECC_FILE_1_NAME       "ca_ecc_cert_1.key.pem"
+#define TEST_CA_PK_ECC_FILE_2_NAME       "ca_ecc_cert_2.key.pem"
+#define TEST_SERVER_PK_ECC_FILE_1_NAME   "server_ecc_cert_1.key.pem"
+#define TEST_SERVER_PK_ECC_FILE_3_NAME   "server_ecc_cert_3.key.pem"
+#define TEST_CLIENT_PK_ECC_FILE_1_NAME   "client_ecc_cert_1.key.pem"
+static char *TEST_CA_PK_ECC_FILE_1       = NULL;
+static char *TEST_CA_PK_ECC_FILE_2       = NULL;
+static char *TEST_SERVER_PK_ECC_FILE_1   = NULL;
+static char *TEST_SERVER_PK_ECC_FILE_3   = NULL;
+static char *TEST_CLIENT_PK_ECC_FILE_1   = NULL;
+
+#define TEST_CHAIN_FILE_PATH_NAME        "chain_file.pem"
+static char *TEST_CHAIN_FILE_PATH        = NULL;
 
 //#############################################################################
 // Test helpers
 //#############################################################################
 
-static void test_helper_setup_homedir(void)
+static void test_helper_setup_temp_dir(char **pp_temp_dir, char **pp_temp_dir_guid)
 {
-    TEST_IOTEDGE_HOMEDIR = hsm_test_util_create_temp_dir(&TEST_IOTEDGE_HOMEDIR_GUID);
-    ASSERT_IS_NOT_NULL_WITH_MSG(TEST_IOTEDGE_HOMEDIR_GUID, "Line:" TOSTRING(__LINE__));
-    ASSERT_IS_NOT_NULL_WITH_MSG(TEST_IOTEDGE_HOMEDIR, "Line:" TOSTRING(__LINE__));
+    char *temp_dir, *guid;
+    temp_dir = hsm_test_util_create_temp_dir(&guid);
+    ASSERT_IS_NOT_NULL_WITH_MSG(guid, "Line:" TOSTRING(__LINE__));
+    ASSERT_IS_NOT_NULL_WITH_MSG(temp_dir, "Line:" TOSTRING(__LINE__));
+    printf("Temp dir created: [%s]\r\n", temp_dir);
+    *pp_temp_dir = temp_dir;
+    *pp_temp_dir_guid = guid;
+}
 
-    printf("Temp dir created: [%s]\r\n", TEST_IOTEDGE_HOMEDIR);
-    hsm_test_util_setenv("IOTEDGE_HOMEDIR", TEST_IOTEDGE_HOMEDIR);
-    printf("IoT Edge home dir set to %s\n", TEST_IOTEDGE_HOMEDIR);
+static void test_helper_teardown_temp_dir(char **pp_temp_dir, char **pp_temp_dir_guid)
+{
+    ASSERT_IS_NOT_NULL_WITH_MSG(pp_temp_dir, "Line:" TOSTRING(__LINE__));
+    ASSERT_IS_NOT_NULL_WITH_MSG(pp_temp_dir_guid, "Line:" TOSTRING(__LINE__));
+
+    char *temp_dir = *pp_temp_dir;
+    char *guid = *pp_temp_dir_guid;
+    ASSERT_IS_NOT_NULL_WITH_MSG(temp_dir, "Line:" TOSTRING(__LINE__));
+    ASSERT_IS_NOT_NULL_WITH_MSG(guid, "Line:" TOSTRING(__LINE__));
+
+    hsm_test_util_delete_dir(guid);
+    free(temp_dir);
+    free(guid);
+    *pp_temp_dir = NULL;
+    *pp_temp_dir_guid = NULL;
 }
 
-static void test_helper_teardown_homedir(void)
+static char* prepare_file_path(const char* base_dir, const char* file_name)
```

Comment:
Seems like this could have been added to hsm_utils.

Reply:
Noted. Will cleanup in a seperate PR.